pgrx/src/datum/into.rs

```diff
@@ -366,26 +366,22 @@ impl<'a> IntoDatum for &'a [u8] {
 
             // SAFETY: `varlena` can properly cast into a `varattrib_4b` and all of what it contains is properly
             // allocated thanks to our call to `palloc` above
-            let varattrib_4b = varlena
-                .cast::<pg_sys::varattrib_4b>()
-                .as_mut()
-                .unwrap_unchecked()
-                .va_4byte
-                .as_mut();
+            let varattrib_4b: *mut _ =
+                &mut varlena.cast::<pg_sys::varattrib_4b>().as_mut().unwrap_unchecked().va_4byte;
```

Comment:
Hm, okay... so we cast the varlena pointer, then call `as_mut()`, then `unwrap_unchecked` it, then deref `.va_4byte`...  since we're just derefing anyways, can't this just be
```rust
&mut (*varlena.cast::<pg_sys::varattrib_4b>()).va_4byte;
```
or something like that?

...also why is it a pointer? If that is important for soundness, this should actually be written with `core::ptr::addr_of_mut!` without the rest. I think when I first reviewed this code, I didn't have my feet under me on the exact semantics of place expressions the way I do now: we want to create a place expression without expressing the strong semantics of `&mut`, which is what the macro I mentioned allows.